daml_dit_ddit/subcommand_build.py

```diff
@@ -140,25 +140,37 @@ def subcommand_main(force: bool, rebuild_dar: bool):
     dar_filename = build_dar(base_filename, rebuild_dar)
     build_pex(tmp_filename)
 
+    subdeployments = (dabl_meta.subdeployments or [])
+
+    resource_files = set()
+
     LOG.info('Enriching output DIT file...')
     with ZipFile(tmp_filename, 'a') as pexfile:
         for pkg_filename in os.listdir('pkg'):
+            resource_files.add(pkg_filename)
             file_bytes = Path(f'pkg/{pkg_filename}').read_bytes()
 
             LOG.info(f'  Adding package file: {pkg_filename}, len=={len(file_bytes)}')
             pexfile.writestr(pkg_filename, file_bytes)
 
         if dar_filename:
             pexfile.write(dar_filename)
+            resource_files.add(dar_filename)
+
+            subdeployments=[*subdeployments, dar_filename]
```

Comment:
This could just be `subdeployments.append(dar_filename)`; you're already mutably modifying `resource_files` in the same loop so there doesn't seem to be an implicit assumption that variables need to be treated immutably here.